$OpenBSD: patch-configure,v 1.1.1.1 2011/09/06 07:10:56 landry Exp $
Remove -Wredundant-decls which makes a lot of useless noise on
OpenBSD because of some problems in system headers.
(+ sndio support)
--- configure.orig	Sun Jul 24 11:42:22 2011
+++ configure	Tue Aug 30 12:28:26 2011
@@ -5,7 +5,7 @@
 check_cflags()
 {
 	check_cc_flag -std=gnu99 -pipe -Wall -Wshadow -Wcast-align -Wpointer-arith \
-		-Wwrite-strings -Wundef -Wmissing-prototypes -Wredundant-decls \
+		-Wwrite-strings -Wundef -Wmissing-prototypes \
 		-Wextra -Wno-sign-compare -Wformat-security
 
 	for i in -Wdeclaration-after-statement \
@@ -276,6 +276,16 @@ check_sun()
 	fi
 }
 
+check_sndio()
+{
+	case `uname -s` in
+	*BSD)
+		check_library SNDIO "" "-lsndio"
+		return $?
+	esac
+	return 1
+}
+
 check_waveout()
 {
 	case `uname -s` in
@@ -365,6 +375,7 @@ Optional Features: y/n
   CONFIG_ARTS     	ARTS                                            [auto]
   CONFIG_OSS      	Open Sound System                               [auto]
   CONFIG_SUN      	Sun Audio                                       [auto]
+  CONFIG_SNDIO		Sndio						[auto]
   CONFIG_WAVEOUT  	Windows Wave Out                                [auto]
   USE_FALLBACK_IP	Use a specific IP for every unrecognized	[n]
                         input format. Currently set to FFMPEG.
@@ -421,6 +432,7 @@ check check_ao      CONFIG_AO
 check check_arts    CONFIG_ARTS
 check check_oss     CONFIG_OSS
 check check_sun     CONFIG_SUN
+check check_sndio   CONFIG_SNDIO
 check check_waveout CONFIG_WAVEOUT
 check check_roar    CONFIG_ROAR
 
@@ -442,6 +454,6 @@ config_header config/iconv.h HAVE_ICONV
 
 makefile_vars bindir datadir libdir mandir exampledir
 makefile_vars CONFIG_FLAC CONFIG_MAD CONFIG_MIKMOD CONFIG_MODPLUG CONFIG_MPC CONFIG_VORBIS CONFIG_WAVPACK CONFIG_WAV CONFIG_MP4 CONFIG_AAC CONFIG_FFMPEG
-makefile_vars CONFIG_ROAR CONFIG_PULSE CONFIG_ALSA CONFIG_AO CONFIG_ARTS CONFIG_OSS CONFIG_SUN CONFIG_WAVEOUT
+makefile_vars CONFIG_ROAR CONFIG_PULSE CONFIG_ALSA CONFIG_AO CONFIG_ARTS CONFIG_OSS CONFIG_SUN CONFIG_SNDIO CONFIG_WAVEOUT
 
 generate_config_mk
